tests/Benchmarks/System.Buffers/Reader_Search.cs

```diff
@@ -22,92 +26,174 @@ public void Setup()
         [Benchmark]
         public void TryReadUntilAny_2_Span()
         {
-            BufferReader reader = new BufferReader(s_rosSplit);
+            BufferReader<byte> reader = new BufferReader<byte>(s_rosSplit);
             byte[] delimiters = { 0, 255 };
-            while (reader.TryReadUntilAny(out ReadOnlySpan<byte> bytes, delimiters))
+            while (reader.TryReadToAny(out ReadOnlySpan<byte> bytes, delimiters))
             {
             }
         }
 
         [Benchmark]
         public void TryReadUntilAny_2_Span_OneSegment()
         {
-            BufferReader reader = new BufferReader(s_ros);
+            BufferReader<byte> reader = new BufferReader<byte>(s_ros);
             byte[] delimiters = { 0, 255 };
-            while (reader.TryReadUntilAny(out ReadOnlySpan<byte> bytes, delimiters))
+            while (reader.TryReadToAny(out ReadOnlySpan<byte> bytes, delimiters))
             {
             }
         }
 
         [Benchmark]
         public void TryReadUntilAny_5_Span()
         {
-            BufferReader reader = new BufferReader(s_rosSplit);
+            BufferReader<byte> reader = new BufferReader<byte>(s_rosSplit);
             byte[] delimiters = { 2, 3, 5, 7, 11 };
-            while (reader.TryReadUntilAny(out ReadOnlySpan<byte> bytes, delimiters))
+            while (reader.TryReadToAny(out ReadOnlySpan<byte> bytes, delimiters))
             {
             }
         }
 
         [Benchmark]
         public void TryReadUntilAny_5_Span_OneSegment()
         {
-            BufferReader reader = new BufferReader(s_ros);
+            BufferReader<byte> reader = new BufferReader<byte>(s_ros);
             byte[] delimiters = { 2, 3, 5, 7, 11 };
-            while (reader.TryReadUntilAny(out ReadOnlySpan<byte> bytes, delimiters))
+            while (reader.TryReadToAny(out ReadOnlySpan<byte> bytes, delimiters))
+            {
+            }
+        }
+
+        [Benchmark]
+        public unsafe void TryReadUntil_2Span()
+        {
+            BufferReader<byte> reader = new BufferReader<byte>(s_rosSplit);
+            byte* b = stackalloc byte[] { 0x0, 0x1 };
+            Span<byte> span = new Span<byte>(b, 2);
+            while (reader.TryReadTo(out ReadOnlySequence<byte> bytes, span))
+            {
+            }
+        }
+
+        [Benchmark]
+        public void TrySkipTo()
+        {
+            BufferReader<byte> reader = new BufferReader<byte>(s_rosSplit);
+            while (reader.TrySkipTo(42))
+            {
+            }
+        }
+
+        [Benchmark]
+        public void TrySkipTo_SingleSegment()
+        {
+            BufferReader<byte> reader = new BufferReader<byte>(s_ros);
+            while (reader.TrySkipTo(42))
+            {
+            }
+        }
+
+        [Benchmark]
+        public unsafe void TrySkipToAny_AndSkipPastAny()
+        {
+            BufferReader<byte> reader = new BufferReader<byte>(s_rosSplit);
+            byte* b = stackalloc byte[] { 0x00, 0xFF };
```

Comment:
Is this type of declaration required by the compiler or can we use safely defined spans here?

Reply:
Required for now- I didn't want to make test specific statics.